x-pack/plugins/cases/public/common/hooks.ts

```diff
@@ -5,11 +5,120 @@
  * 2.0.
  */
 
+import { useCallback, useEffect, useState } from 'react';
```

Comment:
@cnasikas correct me if I'm wrong. I believe these hooks are meant to be exported to other plugins mostly right?

Should we move these changes to `cases/public/containers`?

Also, I believe we use a single file per hook.

Reply:
Yes, I agree. The purpose of this file is to provide hooks for all components. It is better if we co-locate our files. For example, if the hook is used only in the cases table view then it should be in the `x-pack/plugins/cases/public/components/all_cases` like the `useRefreshCases` hook.